### PR TITLE
[Refactor] Add `Syncthing.Fs`

### PR DESCRIPTION
**Pull request by andreafalzetti**
# Proposed changes

This PR goes against this other PR: https://github.com/okteto/okteto/pull/4261

I've kept them into 2 PRs for ease of review. The first one (https://github.com/okteto/okteto/pull/4261) is mainly about the actual change, this one is a refactor to make that change easier.

### CODE REVIEW

File: pkg/cmd/down/down.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+	"os"
+	"os/signal"
+)
+
+type analyticsTrackerInterface interface {
+	TrackDown(bool)
+	TrackDownVolumes(bool)
+}
+
+type Context struct {
```
Comment (ifbyol) [inline pkg/cmd/down/down.go:27]: Why the name `context`? We have so many different contexts that might be confusing (golang context, okteto context, kubernetes context, build context, ....) What does `context` mean in the context of `Down`?

Comment (andreafalzetti) [reply pkg/cmd/down/down.go:27]: @ifbyol I was following the pattern of `upContext`, but I think it's time to find another name :D

Comment (andreafalzetti) [reply pkg/cmd/down/down.go:27]: honestly I cannot think of a better name, despite I think we are absuing Context. Perhaps we can follow-up on this, or if anyone has any suggestion, I am open for feedback cc @maroshii @jLopezbarb

File: pkg/cmd/down/down.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	Dev               *model.Dev
+	Manifest          *model.Manifest
+	Fs                afero.Fs
+	K8sClientProvider okteto.K8sClientProvider
+	AnalyticsTracker  analyticsTrackerInterface
+}
+
+func New(dev *model.Dev, manifest *model.Manifest, fs afero.Fs, k8sClientProvider okteto.K8sClientProvider, at analyticsTrackerInterface) *Context {
```
[the remark this answers is not in the document]
Comment (andreafalzetti) [reply pkg/cmd/down/down.go]: good shout, changed in [20b4207](https://github.com/okteto/okteto/pull/4266/commits/20b4207b492dc23d5829d4a352f8328c626e2315)

**Pull request merged**